### PR TITLE
Support cross compilation of TFLite pip package

### PR DESCRIPTION
**Pull request by vinceab**
This pull request add the support of cross compiling variables.
It allow to cross compile TFLite library and python module for platforms other than aarch64, Rpi and iOS.
It has been successfully tested with Yocto Openembedded environment to build TensorFlow Lite for the new STM32MP1 Microprocessor target from STMicroelectronics.

It also remove some bashisms in pip build script that prevent to build TFLite in server machine with a dash shell. 

BR
Vincent

### CODE REVIEW

**Review by terryheo — COMMENTED** [review]
Could you elaborate in which case bash is not available? Since all the scripts are based on bash, I'm not sure which script should avoid bashism.

**Comment by vinceab** [comment]
Hi,

The build issue has been encounter in different build server on which the host machine has the dash shell set by default.
In that case, the execution of the lite/tools/pip_package/build_pip_package.sh script is failing with the following error:
build_pip_package.sh: 22:  Bad substitution

In this patch, I  am just focusing on the build_pip_package.sh because it is the one that is used from the Yocto environment build system to build the python package.

Vincent

**Comment by terryheo** [comment]
The first line of build_pip_package.sh is "#!/usr/bin/env bash".
Can you install bash on the build server?

> Hi,
> 
> The build issue has been encounter in different build server on which the host machine has the dash shell set by default.
> In that case, the execution of the lite/tools/pip_package/build_pip_package.sh script is failing with the following error:
> build_pip_package.sh: 22: Bad substitution
> 
> In this patch, I am just focusing on the build_pip_package.sh because it is the one that is used from the Yocto environment build system to build the python package.
> 
> Vincent

**Comment by vinceab** [comment]
I understand your point.
I will in that case abandoned the first patch and update the pull request.
Vincent

**Review by terryheo — DISMISSED** [review]
LGTM

**Comment by gbaned** [comment]
@vinceab Can you please address Ubuntu Sanity errors? Thanks!

**Comment by vinceab** [comment]
@gbaned a new patch has been uploaded. that should fix Ubuntu Sanity errors.

**Review by terryheo — DISMISSED** [review]

File: tensorflow/lite/tools/pip_package/build_pip_package.sh [hunk 1]
```diff
@@ -23,7 +23,11 @@ export TENSORFLOW_DIR="${SCRIPT_DIR}/../../../.."
 TENSORFLOW_LITE_DIR="${TENSORFLOW_DIR}/tensorflow/lite"
 TENSORFLOW_VERSION=$(grep "_VERSION = " "${TENSORFLOW_DIR}/tensorflow/tools/pip_package/setup.py" | cut -d= -f2 | sed "s/[ '-]//g")
 export PACKAGE_VERSION="${TENSORFLOW_VERSION}${VERSION_SUFFIX}"
-BUILD_DIR="/tmp/tflite_pip/${PYTHON}"
+if [[ "${TENSORFLOW_CROSS_COMPILATION}" == "true" ]]; then
```
Comment (terryheo) [inline tensorflow/lite/tools/pip_package/build_pip_package.sh]: how to set TENSORFLOW_CROSS_COMPILATION?

Comment (vinceab) [reply tensorflow/lite/tools/pip_package/build_pip_package.sh]: Hi,
TENSORFLOW_CROSS_COMPILATION is a new variable introduce to not break the current way of building Tensorflow pip package.
It should be set by the compilation environment to warn the .sh script that the build directory should change to have a local build directory.

I'd rather like to not have this environment variable but in this case, the BUILD_DIR should be a local build directory for every one meaning BUILD_DIR="$'{SCRIPT_DIR}/gen/tflite_pip" 

What would you prefer?

Comment (terryheo) [reply tensorflow/lite/tools/pip_package/build_pip_package.sh]: What's the issue of using "/tmp/tflite_pip/${PYTHON}" for cross build?

BTW, we also need to update tensorflow/lite/tools/pip_package/Makefile to change it.

Comment (vinceab) [reply tensorflow/lite/tools/pip_package/build_pip_package.sh]: In some cross compilation environment (like Yocto OpenEmbedded), you don't have access the the root directory of your host computer to build. It is impossible to use /tmp/.

Yes your right about the Makefile if you decide to mode the build directory into a local directory.

Do you want me to update my patch taking into account only a local directory for the build?

Comment (terryheo) [reply tensorflow/lite/tools/pip_package/build_pip_package.sh]: I see. Yes, let's use local directory to build which is aligned with lite/tools/make/Makefile behavior. Could you apply it?

Comment (vinceab) [reply tensorflow/lite/tools/pip_package/build_pip_package.sh]: yes for sure, new patch will be available soon

Comment (vinceab) [reply tensorflow/lite/tools/pip_package/build_pip_package.sh]: New patch uploaded

**Review by terryheo — APPROVED** [review]
Thanks for the update.

**Pull request merged**